runtime/jcl/common/shared.c

```diff
@@ -546,40 +543,68 @@ getPathProtocolFromURL(JNIEnv* env, jobject url, jmethodID URLgetPathID, jmethod
 	return rc;
 }
 
+static BOOLEAN
+isPathTypeJimage(const char *pathChars, jsize pathLen)
+{
+	char JimageEndsWith[] = DIR_SEPARATOR_STR "lib" DIR_SEPARATOR_STR "modules";
+	IDATA len = LITERAL_STRLEN(JimageEndsWith);
+
+	if (pathLen >= len) {
+		char *endsWith = (char*)(pathChars + (pathLen - len));
+		if (strncmp(endsWith, JimageEndsWith, len) == 0) {
+			return TRUE;
+		}
+	}
+	return FALSE;
+}
 
 /* THREADING: Can be called multi-threaded */
 static UDATA
-getCpeTypeForProtocol(char* protocol, jsize protocolLen, const char* pathChars, jsize pathLen)
+getCpeTypeForProtocol(JNIEnv *env, char *protocol, jsize protocolLen, const char *urlPathChars, jsize urlPathLen, char **correctedPathPtr)
 {
+	J9JavaVM *vm = ((J9VMThread*)env)->javaVM;
+	const char *pathChars = NULL;
+	jsize pathLen = 0;
+
 	Trc_JCL_com_ibm_oti_shared_getCpeTypeForProtocol_Entry();
 
 	if (!protocol) {
-		Trc_JCL_com_ibm_oti_shared_getCpeTypeForProtocol_Exit0();
-		return 0;
+		Trc_JCL_com_ibm_oti_shared_getCpeTypeForProtocol_ExitFail1();
+		return CPE_TYPE_UNKNOWN;
+	}
+	if (!correctURLPath(env, urlPathChars, urlPathLen, correctedPathPtr, &(vm->sharedClassConfig->jclStringFarm))) {
+		Trc_JCL_com_ibm_oti_shared_getCpeTypeForProtocol_ExitFail2();
+		return CPE_TYPE_UNKNOWN;
 	}
+	pathChars = *correctedPathPtr;
+	pathLen = (jsize)strlen(pathChars);
 	if (strncmp(protocol, "jar", 4)==0) {
```

Comment:
Constant belongs on the left, spaces both sides of `==`:
```c
	if (0 == strncmp(protocol, "jar", 4)) {
```
Similar comment for line 585.